### PR TITLE
Use jobs order provided by the cloud

### LINKED ISSUE #570: Adopt built-in job ordering from `pasqal-cloud 0.3.5`
`pasqal-cloud 0.3.5` added job ordering to the batch, so now we can get rid of how we are conserving the job order in the `PasqalCloud` and rely on `Batch.ordered_jobs` instead.

### PR DESCRIPTION
**Pull request by HGSilveri**
Closes #570 .

### CODE REVIEW

File: tests/test_pasqal.py [hunk 1]
```diff
@@ -208,16 +208,6 @@ def test_submit(fixt, parametrized, emulator, seq, mock_job):
...
-    job_params[0]["runs"] = 1
-    with pytest.raises(RuntimeError, match="Failed to find job ID"):
-        # Job runs don't match MockJob
-        fixt.pasqal_cloud.submit(
-            seq,
-            job_params=job_params,
-            emulator=emulator,
-            config=config,
-        )
-
```
Comment (a-corni) [inline tests/test_pasqal.py:220]: Why is this test no longer relevant ?

Comment (HGSilveri) [reply tests/test_pasqal.py:220]: Because we no longer try to determine the jobs order (the code where this was being raised was deleted)

Comment (a-corni) [reply tests/test_pasqal.py:220]: Okay, that was my guess... So it's no longer an issue that the number of runs in job_params is different than number of runs in "mock_job" ? The two issues seemed independent to me, but I am not I understood the test in the first place...

Comment (HGSilveri) [reply tests/test_pasqal.py:220]: That manual change was just a way for the job params to not match anymore and thus trigger the fail to find the job ID.

**Pull request merged**